app/components/FrameworkSelect.tsx

```diff
@@ -0,0 +1,118 @@
+import { Fragment } from 'react'
+import { Listbox, Transition } from '@headlessui/react'
+
+import { HiCheck, HiChevronDown } from 'react-icons/hi'
+
+import vueLogo from '../images/vue-logo.svg'
+import reactLogo from '../images/react-logo.svg'
+import solidLogo from '../images/solid-logo.svg'
+import svelteLogo from '../images/svelte-logo.svg'
+import angularLogo from '../images/angular-logo.svg'
+import { Form, useMatches } from '@remix-run/react'
+import { useNavigate } from '@remix-run/react'
+
+// TODO: we could potentially pass available frameworks trough config
+const frameworks = [
+  { label: 'React', value: 'react' },
+  { label: 'Solid', value: 'solid' },
+  { label: 'Vue', value: 'vue' },
+  { label: 'Svelte', value: 'svelte' },
+  { label: 'Angular', value: 'angular' },
+]
+
+const logoMap = {
+  react: reactLogo,
+  solid: solidLogo,
+  vue: vueLogo,
+  svelte: svelteLogo,
+  angular: angularLogo,
+}
+
+const properCase = (str: string) => str[0].toUpperCase() + str.slice(1)
+
+export function FrameworkSelect() {
+  const navigate = useNavigate()
+  const matches = useMatches()
+  
+  const match = matches.find(m => m.id === 'routes/query/v4/docs/$framework')
```

Comment:
given that we are in `app/components/FrameworkSelect.tsx` here, is it correct to have a hardcoded match upon query v4 here?

Reply:
Framework is passed from upper scope via prop now.